web/js/map-OpenLayers.js

```diff
@@ -233,7 +234,7 @@ $.extend(fixmystreet.utils, {
             }
             var id = pin[3] === undefined ? pin[3] : +pin[3];
             var marker_size = (id === window.selected_problem_id) ? selected_size : size;
-            var draggable = (id === window.selected_problem_id) ? true : (pin[6] === false ? false : true);
+            var draggable = (pin[6] === undefined && id === window.selected_problem_id) ? true : (pin[6] === false ? false : true);
```

Comment:
I don't think pin[6] can ever be undefined any more, it's always set to a boolean by map.js. So I think this breaks the behaviour where you e.g. go to an /reports all reports page, click a report, that pin becomes draggable, pan the map a bit, markers reload, this function will be called, and it won't be draggable any more.
Was there something being fixed by adding this? I think the behaviour should be okay if left as it was?

Reply:
So I think what was happening is that `id === window.selected_problem_id` was `true` when both were `undefined`, meaning the pin was draggable even if `pin[6]` was `false`. This was an attempt to fix that, but perhaps there's a better approach?

Reply:
`id && id === window.selected_problem_id`?

Reply:
🤦 seems so obvious I'm not sure why I didn't try that already! Will give it a go, thanks.